src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -281,8 +281,13 @@ internal int[] Sort(TElement[] elements, int count, int minIdx, int maxIdx)
             return map;
         }
 
-        internal TElement ElementAt(TElement[] elements, int count, int idx) =>
-            elements[QuickSelect(ComputeMap(elements, count), count - 1, idx)];
+        internal TElement ElementAt(TElement[] elements, int count, int idx)
+        {
+            int[] map = ComputeMap(elements, count);
+            return idx == 0 ?
```

Comment:
It's unlikely to affect anything measurably, but in the name of completeness, you've validated that this extra check / the extra code in the method doesn't negatively impact cases beyond the positive cases you created benchmarks for?